### ISSUE TITLE
Just a question

### ISSUE THREAD

**Issue by deccosantos** [nickfox/GpsTracker#96]
Hi everyone, i'm using this server and this is great, but i'm facing some problems and i didn't find any answer even in stackoverflow.

So my problem is that now i have about 150 vehicles in the server, and after some time the server hangs and stop to send responses back to client and after some time closes port.

I'm using xampp in a windows server 2012 I just want to know if, and what, changes in php.ini and apache.conf should i do to keep alive server? 

I think that the problem is in fread, because if a set fread to 128 as example server hangs fastly than if a set fread to 1024 or 2048.

Since now thanks to all who helps me!

**Comment by nickfox** [comment]
Have you created indexes in the database?

**Comment by deccosantos** [comment]
@nickfox no, but I'm not doing a select from database, just insert, even so i need to create indexes? Remembering that when server hangs no erro is shown in log even in mysql log

**Comment by brentfraser** [comment]
check the web server log (apache?)

**Comment by brentfraser** [comment]
Perhaps PHP is not a good choice for this socket-oriented server script.

**Comment by brentfraser** [comment]
It could be that one of the vehicle devices has not implemented the protocol properly.  Are they all the same?

**Comment by deccosantos** [comment]
in the apache log there is no erros, all the trackers are the same, the problem maybe is with the fread or something in the config of php

**Comment by deccosantos** [comment]
i know php shoukd not be the best, but it works for nick

**Comment by brentfraser** [comment]
We haven't tested with 150 devices so feel free to use s different language.  I saw Stack overflow message saying Node.js has good socket functions. https://stackoverflow.com/questions/23741634/php-socket-server-hangs

**Comment by nickfox** [comment]
150 should not be a problem. Here are a few things to look at:

https://stackoverflow.com/questions/9819271/why-is-mysql-innodb-insert-so-slow

http://www.ajaydivakaran.com/mysql-innodb-are-inserts-slowing-down/

https://www.quora.com/Why-would-an-%E2%80%9CINSERT%E2%80%9D-query-e-g-indexes-table-locking-etc-be-slow-in-MySQL

n

**Comment by deccosantos** [comment]
@nickfox the problem is not with database or mysql... I remove from my server all mysql functions and still hangs...

**Comment by nickfox** [comment]
Try a different server. I use nginx on ubuntu. It is hosted by linode.com. Or else find someone with expertise using the server you have.

**Comment by nickfox** [comment]
Did you figure it out?

n

**Comment by deccosantos** [comment]
@nickfox i changed all my server to Nodejs and the problem never more happens, this was my resolution!

**Comment by nickfox** [comment]
Excellent. Good job.

n

**Comment by Helen300** [comment]
Could you show us how you changed all your server to nodejs?  @deccosantos
